tests/ui/lifetimes/issue-107988.rs

```diff
@@ -0,0 +1,16 @@
+pub trait TraitEngine<'tcx>: 'tcx {}
+
+pub trait TraitEngineExt<'tcx> {
+    fn register_predicate_obligations(&mut self);
+}
+
+impl<T: ?Sized + TraitEngine<'tcx>> TraitEngineExt<'tcx> for T {
+  //~^ HELP consider introducing lifetime `'tcx` here
+  //~| ERROR use of undeclared lifetime name `'tcx`
+  //~| consider making the bound lifetime-generic with a new `'tcx` lifetime
+  //~| ERROR use of undeclared lifetime name `'tcx`
+  //~| HELP consider introducing lifetime `'tcx` here
```

Comment:
Typically you don't need to incldue all of the `HELP` messages unless the UI test is actually checking for those on purpose.

```suggestion
  //~^ ERROR use of undeclared lifetime name `'tcx`
  //~| ERROR use of undeclared lifetime name `'tcx`
```